modules/avro-kafka-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/kafka/AvroKafkaSpanCompleter.scala

```diff
@@ -29,17 +29,18 @@ object AvroKafkaSpanCompleter {
       completer <- QueuedSpanCompleter[F](process, exporter, bufferSize, batchSize, batchTimeout)
     } yield completer
 
-  def fromProducer[F[_]: ConcurrentEffect: ContextShift: Timer](
+  def fromProducer[F[_]: ConcurrentEffect: Timer](
     process: TraceProcess,
     producer: KafkaProducer[F, TraceId, CompletedSpan],
     topic: String,
     bufferSize: Int = 2000,
     batchSize: Int = 50,
     batchTimeout: FiniteDuration = 10.seconds
-  ): Resource[F, SpanCompleter[F]] =
+  ): Resource[F, SpanCompleter[F]] = {
+    val exporter = AvroKafkaSpanExporter.fromProducer[F, Chunk](producer, topic)
```

Comment:
This reorder was made to remove a false-positive `unused` warning related to the `better-monadic-for`'s rewriting.

Reply:
Nice!